clickhouse-admin/src/context.rs

```diff
@@ -36,6 +60,10 @@ impl KeeperServerContext {
     pub fn log(&self) -> &Logger {
         &self.log
     }
+
+    pub async fn generation(&self) -> Option<Generation> {
+        *self.generation.lock().await
```

Comment:
We only need read access here, and so we can easily avoid an async mutex here. `Generation` is also `Copy`, so this is cheap. I'd suggest making this a synchronous function and  calling `*self.generation.lock()` instead.

Reply:
Yeah, I was wrong here. I wasn't considering the usage of the generation with regards to concurrent requests.